File: algorithms/rs/283.移动零.rs
/*
 * @lc app=leetcode.cn id=283 lang=rust
 *
 * [283] 移动零
 */
// @lc code=start
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut i = 0;
        for j in 0..nums.len() {
            if (nums[j] != 0) {
                nums[i] = nums[j];
                i += 1;
            }
        }
        while i < nums.len() {
            nums[i] = 0;
            i += 1;
        }
    }
}
// @lc code=end

File: algorithms/golang/283.移动零.go
/*
 * @lc app=leetcode.cn id=283 lang=golang
 *
 * [283] 移动零
 */

// @lc code=start
func moveZeroes(nums []int) {
	for p, i := 0, 0; i < len(nums); i++ {
		if nums[i] != 0 {
			nums[i], nums[p] = nums[p], nums[i]
			p++
		}
	}
}

// @lc code=end
